Java/src/test/java/com/thealgorithms/dynamicprogramming/RegexMatchingTest.java

```diff
@@ -0,0 +1,130 @@
+            var inputPrefix = inputString.split(" ")[0];
+            list.add(DynamicTest.dynamicTest(
+                    "RegexRecursion Method 1 - " + patternName + ": " + inputPrefix,
+                    () -> assertMethod.accept(RegexMatching.regexRecursion(inputString, wildcardPattern))));
+        });
+    }
+
+    void addMethod2ToList(List<DynamicTest> list, String wildcardPattern, String patternName, Consumer<Boolean> assertMethod) {
+        Arrays.stream(TEST_STRINGS).forEach(inputString -> {
+            var inputPrefix = inputString.split(" ")[0];
+            list.add(DynamicTest.dynamicTest(
+                    "RegexRecursion Method 2 - " + patternName + ": " + inputPrefix,
+                    () -> assertMethod.accept(RegexMatching.regexRecursion(inputString, wildcardPattern, 0, 0))));
+        });
+    }
+
+    void addMethod3ToList(List<DynamicTest> list, String wildcardPattern, String patternName, Consumer<Boolean> assertMethod) {
+        Arrays.stream(TEST_STRINGS).forEach(inputString -> {
+            var inputPrefix = inputString.split(" ")[0];
+            list.add(DynamicTest.dynamicTest(
+                    "RegexRecursion Method 3 - " + patternName + ": " + inputPrefix,
+                    () -> assertMethod.accept(RegexMatching.regexRecursion(
+                            inputString,
+                            wildcardPattern,
+                            0,
+                            0,
+                            new int[inputString.length()][wildcardPattern.length()])
+                    )
+            ));});
+    }
+
+    void addMethod1ToList(List<DynamicTest> list, Pair<String, String> input, String patternName, Consumer<Boolean> assertMethod) {
+        list.add(DynamicTest.dynamicTest(
+                    "RegexRecursion Method 1: " + patternName,
+                    () -> assertMethod.accept(RegexMatching.regexRecursion(input.first, input.second))
+                )
+        );
+    }
+
+    void addMethod2ToList(List<DynamicTest> list, Pair<String, String> input, String patternName, Consumer<Boolean> assertMethod) {
+            list.add(DynamicTest.dynamicTest(
+                    "RegexRecursion Method 2: " + patternName,
+                    () -> assertMethod.accept(RegexMatching.regexRecursion(input.first, input.second, 0, 0))));
+    }
+
+    void addMethod3ToList(List<DynamicTest> list, Pair<String, String> input, String patternName, Consumer<Boolean> assertMethod) {
+        list.add(DynamicTest.dynamicTest(
+                "RegexRecursion Method 3: " + patternName,
+                () -> assertMethod.accept(RegexMatching.regexRecursion(
+                        input.first,
+                        input.second,
+                        0,
+                        0,
+                        new int[input.first.length()][input.second.length()])
+                )
+        ));
+    }
+
+    record Pair<T, Y>(T first, Y second) {}
+}
```

Comment:
Perhaps we should avoid adding the improved coverage unit-tests to the `develop` branch. Keep these in the `improved-coverage` branch instead

Reply:
These are just here to allow for running and showing that the coverage count works. I had to do this since there just weren't any tests for this class, so I cherry-picked it from the other branch. This one isn't really a proper test though in the sense of a unit test, since it only checks what has been run through. I'd believe it's necessary for some level of this because of the part of the issue: 

> At the end of the program (as the last instruction in “main” or at the end of all unit tests), write all information about the taken branches taken to a file or console.